ms2deepscore/models/SiameseModel.py

```diff
@@ -98,20 +98,22 @@ def save(self, filename: Union[str, Path]):
 
     @staticmethod
     def _get_base_model(input_dim: int,
-                        dims: Tuple[int, int, int] = (600, 500, 500),
+                        dims: Tuple[int, ...] = (600, 500, 500),
                         embedding_dim: int = 400,
                         dropout_rate: float = 0.25):
         model_input = keras.layers.Input(shape=input_dim, name='base_input')
-        embedding = keras.layers.Dense(dims[0], activation='relu', name='dense1',
-                                       kernel_regularizer=keras.regularizers.l1_l2(l1=1e-6, l2=1e-6))(model_input)
-        embedding = keras.layers.BatchNormalization(name='normalization1')(embedding)
-        embedding = keras.layers.Dropout(dropout_rate, name='dropout1')(embedding)
-        embedding = keras.layers.Dense(dims[1], activation='relu', name='dense2')(embedding)
-        embedding = keras.layers.BatchNormalization(name='normalization2')(embedding)
-        embedding = keras.layers.Dropout(dropout_rate, name='dropout2')(embedding)
-        embedding = keras.layers.Dense(dims[2], activation='relu', name='dense3')(embedding)
-        embedding = keras.layers.BatchNormalization(name='normalization3')(embedding)
-        embedding = keras.layers.Dropout(dropout_rate, name='dropout3')(embedding)
+        for i, dim in enumerate(dims):
+            if i == 0:
+                embedding = keras.layers.Dense(dim, activation='relu', name='dense'+str(i+1),
+                                               kernel_regularizer=keras.regularizers.l1_l2(l1=1e-6, l2=1e-6))(
+                   model_input)
+            else:
+                embedding = keras.layers.Dense(dim, activation='relu', name='dense'+str(i+1),
+                                               kernel_regularizer=keras.regularizers.l1_l2(l1=1e-6, l2=1e-6))(
+                   embedding)
+            embedding = keras.layers.BatchNormalization(name='normalization'+str(i+1))(embedding)
+            embedding = keras.layers.Dropout(dropout_rate, name='dropout'+str(i+1))(embedding)
+
         embedding = keras.layers.Dense(embedding_dim, activation='relu', name='embedding')(
             embedding)
```

Comment:
Only the last layer I would call 'embedding', maybe `x` seems more appropriate to assign the intermediate layers to. Also the if/else statement is not necessary I think.
```suggestion
        x = model_input
        for i, dim in enumerate(dims):
                x = keras.layers.Dense(dim, activation='relu', name='dense'+str(i+1),
                                               kernel_regularizer=keras.regularizers.l1_l2(l1=1e-6, l2=1e-6))(
                   x)
            x = keras.layers.BatchNormalization(name='normalization'+str(i+1))(x)
            x = keras.layers.Dropout(dropout_rate, name='dropout'+str(i+1))(x)

        embedding = keras.layers.Dense(embedding_dim, activation='relu', name='embedding')(x)
```

Reply:
I also wasn't fully happy with the if-else, but the input layer needed a different name for the final construction of the `Model` in `return keras.Model(model_input, embedding, name='base')`.

Reply:
But changing all other `embedding` to `x` is fine for me (the linter might complain because it doesn't like 1-character variables, but I can check that).